### PR TITLE
RHINENG-4700_and_5552: Use deletion for culling and rename attributes in custom staleness

### PR DESCRIPTION
**Pull request by thearifismail**
# Overview

This PR is being created to address [RHINENG-4700](https://issues.redhat.com/browse/RHINENG-4700) and [RHINENG-5552](https://issues.redhat.com/browse/RHINENG-5552) and replace the [PR #1550](https://github.com/RedHatInsights/insights-host-inventory/pull/1550) 

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [x] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by fellipeh — CHANGES_REQUESTED** [review]
How this will work with the current databases? I saw the changes on migrations folder only for the fields name creation..,

**Comment by thearifismail** [comment]
> How this will work with the current databases? I saw the changes on migrations folder only for the fields name creation..,

After making these changes, I successfully ran the make task `make upgrade_db` followed by `pytest`.

**Review by fellipeh — APPROVED** [review]
lgtm

**Comment by fstavela** [comment]
> > How this will work with the current databases? I saw the changes on migrations folder only for the fields name creation..,
> 
> After making these changes, I successfully ran the make task `make upgrade_db` followed by `pytest`.

Did you run the DB upgrade on an already existing database or on a new one? This change to the DB migration is my only concern with this PR, I think the proper way would be to create a new migration instead of updating an existing one. I don't know if this approach will work on existing databases in Stage and Prod. But I've never done this, so correct me if I'm wrong.

Otherwise, the PR looks good.

**Comment by thearifismail** [comment]
> > > How this will work with the current databases? I saw the changes on migrations folder only for the fields name creation..,
> > 
> > 
> > After making these changes, I successfully ran the make task `make upgrade_db` followed by `pytest`.
> 
> Did you run the DB upgrade on an already existing database or on a new one? This change to the DB migration is my only concern with this PR, I think the proper way would be to create a new migration instead of updating an existing one. I don't know if this approach will work on existing databases in Stage and Prod. But I've never done this, so correct me if I'm wrong.
> 
> Otherwise, the PR looks good.

Thank you @fstavela; very good point.  I ran it on a new DB but let me run against an older one, i.e. migration.

File: api/staleness_query.py [hunk 1]
```diff
@@ -38,12 +38,12 @@ def _build_staleness_sys_default(org_id, config=None):
...
-            "conventional_culling_delta": config.conventional_culling_seconds,
-            "immutable_staleness_delta": config.immutable_staleness_seconds,
-            "immutable_stale_warning_delta": config.immutable_stale_warning_seconds,
-            "immutable_culling_delta": config.immutable_culling_seconds,
+            "conventional_time_to_stale": config.conventional_staleness_seconds,
+            "conventional_time_to_stale_warning": config.conventional_stale_warning_seconds,
+            "conventional_time_to_delete": config.conventional_culling_seconds,
+            "immutable_time_to_stale": config.immutable_staleness_seconds,
+            "immutable_time_to_stale_warning": config.immutable_stale_warning_seconds,
+            "immutable_time_to_delete": config.immutable_culling_seconds,
```
Comment (kruai) [inline api/staleness_query.py:46]: We might as well update the field names in the config file and environment variables too; it'll be confusing if we don't.

**Review by kruai — CHANGES_REQUESTED** [review]
+1 to the point about the migrations - you'll need to make a new migration rather than modify the existing ones.
Also, there are a few places where the field names haven't been updated, and it would make sense to update them all at once.

**Comment by thearifismail** [comment]
> +1 to the point about the migrations - you'll need to make a new migration rather than modify the existing ones. Also, there are a few places where the field names haven't been updated, and it would make sense to update them all at once.

hmmm... I did bulk find and resplace.  Let me look.  Thanks.

File: api/staleness_query.py (hunk 1 above)
Comment (thearifismail) [reply api/staleness_query.py]: Done!!!

**Review by jpramos123 — APPROVED** [review]
LGTM!

Thanks for this PR.

the `_time_to` was a good approach to solve the variables naming we discussed!!

**Pull request merged**